### PR TITLE
Issue #6459: Allow "chunked" Transfer-Encoding when returning Image Style Derivatives.

### LINKED ISSUE #6459: Fixed: All PRs failing `ImageStylesPathAndUrlUnitTest`.
## Description of the bug

About 3 days ago all PRs started failing (there's only been 3 PRs in that time). The first one that failed seems to be https://github.com/backdrop/backdrop/pull/4696

The failures are coming from `ImageStylesPathAndUrlUnitTest` with the error:

```
 Fail      Other      image.test        249     _testImageStyleUrlAndPath()                                             
    Expected Content-Length was reported.
```

image

It happens on PHP 5.6, 7.4, and 8.3 versions.

## Steps To Reproduce

To reproduce the behavior:
1. File any PR with any change, even just the README.md
2. The PR will fail with the above test failing.

Add any other information that could help, such as:
- Backdrop CMS version: 1.x
- Web server and its version: Seemingly Apache on GHA
- PHP version: All
- Database sever (MySQL or MariaDB?) and its version: 10.3
- Operating System and its version: Ubuntu
- Browser(s) and their versions: N/A

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/6459

Allows Apache "chunked" responses as valid in `ImageStylesPathAndUrlUnitTest`

### CODE REVIEW

File: core/modules/image/tests/image.test [hunk 1]
```diff
@@ -114,8 +125,6 @@ class ImageFieldTestCase extends BackdropWebTestCase {
  */
 class ImageStylesPathAndUrlUnitTest extends BackdropWebTestCase {
   protected $style_name;
-  protected $image_info;
-  protected $image_filepath;
```
Comment (quicksketch) [inline core/modules/image/tests/image.test:118]: Just a little code cleanup, these properties are declared but never assigned or used.

File: core/modules/image/tests/image.test [hunk 2]
```diff
@@ -246,7 +258,23 @@ class ImageStylesPathAndUrlUnitTest extends BackdropWebTestCase {
     $this->assertRaw(file_get_contents($generated_uri), 'URL returns expected file.');
```
Comment (quicksketch) [inline core/modules/image/tests/image.test:258]: This line already verifies that the returned response matches the generated image contents.

File: core/modules/image/tests/image.test [hunk 3]
```diff
@@ -265,13 +293,13 @@ class ImageStylesPathAndUrlUnitTest extends BackdropWebTestCase {
...
-      $this->assertFalse(file_exists($generated_uri_noaccess), 'Generated file does not exist.');
-      $generate_url_noaccess = image_style_url($this->style_name, $original_uri_noaccess);
+      $file_no_access = array_shift($files);
+      $original_uri_no_access = file_unmanaged_copy($file_no_access->uri, $scheme . '://', FILE_EXISTS_RENAME);
+      $generated_uri_no_access = $scheme . '://styles/' . $this->style_name . '/' . $scheme . '/'. backdrop_basename($original_uri_no_access);
+      $this->assertFalse(file_exists($generated_uri_no_access), 'Generated file does not exist.');
+      $generate_url_no_access = image_style_url($this->style_name, $original_uri_no_access);
 
-      $this->backdropGet($generate_url_noaccess);
+      $this->backdropGet($generate_url_no_access);
```
Comment (klonos) [inline core/modules/image/tests/image.test:302]: Love this cleanup here ...assuming that CSpell complained about the `noaccess` word.

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

File: core/modules/image/tests/image.test [hunk 4]
```diff
@@ -246,7 +258,23 @@ class ImageStylesPathAndUrlUnitTest extends BackdropWebTestCase {
...
+    $content_length = $this->backdropGetHeader('Content-Length');
+    $transfer_encoding = $this->backdropGetHeader('Transfer-Encoding');
+    if ($transfer_encoding == 'chunked') {
+      $this->pass('Response was chunked without a Content-Length.');
+    }
+    elseif ($content_length == $generated_image_info['file_size']) {
+      $this->pass('Expected Content-Length was reported.');
+    }
+    else {
+      $this->fail('Expected Content-Length or Chunked response was not returned');
```
Comment (quicksketch) [inline core/modules/image/tests/image.test]:
```suggestion
      $this->fail('Expected Content-Length or Chunked response was not returned.');
```

**Comment by quicksketch** [comment]
Merged into 1.x and 1.27.x. Thanks @indigoxela and @klonos for reviewing!

**Pull request merged**